### PR TITLE
Refactor: convert argc/argv to Strings

### PR DESCRIPTION
**Pull request by WalterBright**
Replace some very old code that manually managed memory with Strings. Reduction in complexity and lines of code.

### CODE REVIEW

**Comment by WalterBright** [comment]
@yebblies halp, I don't know how to get this to work with magicport.

**Comment by yebblies** [comment]
> `root/response.d(14): Error: template instance Array!(const(char)*) template 'Array' is not defined`

The `typedef Strings` line will cause the code `alias Strings = Array!(const(char)*);` to be added to `response.d`, but it needs to import `root.array` to get access to `Array`.

Alternatively, you could instead add an import of `root.filename` to response's imports list, because `root.filename` already defines `Strings`.

**Comment by WalterBright** [comment]
ok, trying your advice!

**Comment by yebblies** [comment]
Looks like it worked!

File: src/magicport.json [hunk 1]
```diff
@@ -3198,12 +3198,11 @@
                 "core.stdc.stdio",
                 "core.stdc.stdlib",
                 "core.stdc.string",
+		"root.filename",
```
Comment (yebblies) [inline src/magicport.json]: Alignment

File: src/link.c [hunk 2]
```diff
@@ -842,7 +842,7 @@ int runProgram()
     //printf("runProgram()\n");
     if (global.params.verbose)
     {
-        fprintf(global.stdmsg, "%s", global.params.exefile);
+        //fprintf(global.stdmsg, "%s", global.params.exefile);
```
Comment (yebblies) [inline src/link.c]: Should be removed

**Comment by WalterBright** [comment]
Thanks! Pull it quick!

**Comment by yebblies** [comment]
Auto-merge toggled on

**Pull request merged**